app/src/validators/user.js

```diff
@@ -0,0 +1,75 @@
+const { validate, Joi } = require('express-validation');
+
+const schema = {
+  searchUsers: {
+    query: Joi.object({
+      userId: Joi.alternatives().try(
+        Joi.array().items(Joi.string().guid({
+          version: 'uuidv4'
+        })),
+        Joi.string().guid({
+          version: 'uuidv4'
+        })
+      ),
+      identityId: Joi.alternatives().try(
+        Joi.array().items(Joi.string().guid({
+          version: 'uuidv4'
+        })),
+        Joi.string().guid({
+          version: 'uuidv4'
+        })
+      ),
+      idp: Joi.alternatives().try(
+        Joi.array().items(Joi.string().max(255)),
+        Joi.string().max(255)
+      ),
+      username: Joi.string().pattern(/^[\w\-\s]+$/).max(255),
+      email: Joi.string().max(255).email(),
+      firstName: Joi.string().pattern(/^[\w\-\s]+$/).max(255),
+      fullName: Joi.string().pattern(/^[\w\-\s]+$/).max(255),
+      lastName: Joi.string().pattern(/^[\w\-\s]+$/).max(255),
```

Comment:
We should have `username`, `firstName` and `lastName` use the `string.alphaNum()` restriction; the only one that should have the regex is fullName as we need to allow alphanumeric strings with one or more spaces.